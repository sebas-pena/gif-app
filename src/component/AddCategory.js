import React, { useState } from 'react'
import PropTypes from 'prop-types'



export const AddCategory = ({ setCategories, categories }) => {

    const [inputValue, setInputValue] = useState('')

    const handleSubmit = (e) => {
        e.preventDefault()
        if (inputValue && !categories.includes(inputValue)) {
            setCategories([inputValue, ...categories])
            setInputValue('')
        }
    }

    const handleInputChange = (e) => setInputValue(e.target.value)


    return (
        <form onSubmit={handleSubmit} >
            <input
                type="text"
                value={inputValue}
                onChange={handleInputChange}
                name='category'
                placeholder='Add Category'
            />
        </form>
    )

}

AddCategory.propTypes = {
    setCategories: PropTypes.func.isRequired
}
