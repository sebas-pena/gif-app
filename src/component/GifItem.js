import React from 'react'
import PropTypes from 'prop-types'
export const GifItem = ({ url, title }) => {
    return (
        <div className='card animate__animated animate__fadeIn'>
            <img src={url} alt={title} />
            <p>{title}</p>
        </div>
    )
}

GifItem.propTypes = {
    url: PropTypes.string.isRequired,
    title: PropTypes.string.isRequired,
}