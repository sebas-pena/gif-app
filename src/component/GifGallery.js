import React from 'react'
import { useFetchGifs } from '../hooks/useFetchGifs'
import { GifItem } from './GifItem'
import PropTypes from 'prop-types'

export const GifGallery = ({ category }) => {

    const { data, isLoading } = useFetchGifs(category)

    return (
        <div className=' animate__animated animate__fadeIn' >
            <h3>{category}</h3>
            <div className='gif-gallery'>
                {isLoading && <p>Loading...</p>}
                {
                    data.map(gif => <GifItem key={gif.id} {...gif} />)
                }
            </div>
            <hr />
        </div>
    )
}


GifGallery.propTypes = {
    category: PropTypes.string.isRequired
}