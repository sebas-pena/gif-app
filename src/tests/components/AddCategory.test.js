import React from 'react'
import { shallow } from 'enzyme'
import { AddCategory } from '../../component/AddCategory'


describe('Pruebas sobre <AddCategory />', () => {

    const setCategories = jest.fn()
    const wrapper = shallow(<AddCategory setCategories={setCategories} categories={['']} />)

    beforeEach(() => {
        jest.clearAllMocks()
    })


    test('Debe de mostrarse correctamente', () => {

        expect(wrapper).toMatchSnapshot()

    })

    test('El input debe coincidir con lo tipado', () => {
        const input = wrapper.find('input')
        input.simulate("change", { target: { value: "foo" } })
    })

    test('NO debe postear la informacion con el input vacio', () => {

        wrapper.find('form').simulate('submit', { preventDefault() { }, target: {} })

        expect(setCategories)

    })

    test('Debe postear la informacion y vaciar el input', () => {

        const input = wrapper.find('input')

        input.simulate('change', { target: { value: 'hola' } })

        wrapper.find('form').simulate('submit', { preventDefault() { }, target: {} })

        expect(setCategories).toHaveBeenCalledTimes(1)
        expect(input.prop('value')).toBe('')

    })

})