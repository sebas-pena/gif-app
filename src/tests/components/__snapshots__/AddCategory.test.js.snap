// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Pruebas sobre <AddCategory /> Debe de mostrarse correctamente 1`] = `
<form
  onSubmit={[Function]}
>
  <input
    name="category"
    onChange={[Function]}
    placeholder="Add Category"
    type="text"
    value=""
  />
</form>
`;
